addons/udes_stock/report/deliveryslip_templates.xml

```diff
@@ -7,147 +7,139 @@
-                                <th><strong>Trailer unit ID</strong></th>
-                                <th><strong>Vehicle registration</strong></th>
-                                <th><strong>Driver name</strong></th>
-                            </tr>
-                        </thead>
-                        <tbody>
-                            <tr>
-                                <td>
-                                    <span t-field="o.u_vehicle_sequence"/>
-                                </td>
-                                <td>
-                                    <span t-field="o.u_vehicle_description"/>
-                                </td>
-                                <td>
-                                    <span t-field="o.u_vehicle_registration"/>
-                                </td>
-                                <td>
-                                    <span t-field="o.u_driver_name"/>
-                                </td>
-                            </tr>
-
-                            <t t-if="o.partner_id">
-                                <tr>
-                                    <th colspan="2" style="padding-top: 40px" />
-                                    <th colspan="2" style="padding-top: 40px"><strong>Customer address</strong></th>
-                                </tr>
-                                <tr>
-                                    <td colspan="2" />
-                                    <td colspan="2">
-                                        <span t-field="o.partner_id"
-                                           t-options='{"widget": "contact", "fields": ["address", "name", "phone"], "no_marker": True}'/>
-                                    </td>
-                                </tr>
-                            </t>
+                <table class="table" style="border:0px solid transparent;">
+                    <tbody>
+                        <tr style="border:0px solid transparent;">
+                            <td style="border:0px solid transparent;">
+                                <strong>Reference: </strong>
+                                <span t-field="o.name"/>
+                            </td>
+                            <td style="border:0px solid transparent;">
+                                <span style="padding:0 0px 0 50%;">
+                                <strong>Date: </strong>
+                                <t t-if="o.state == 'done'">
+                                    <span t-esc="o.date_done.strftime('%d-%m-%Y %H:%M')"/>
+                                </t>
+                                <t t-if="o.state != 'done'">
+                                    <span t-esc="o.scheduled_date.strftime('%d-%m-%Y %H:%M')"/>
+                                </t>
+                                </span>
+                            </td>
+                        </tr>
+                    </tbody>
+                </table>
```

Comment:
This is the only change I made in the new commit so that picking reference name and date are displayed correctly as per reference document on the story.